### DISCUSSION TITLE
Run Theia IDE in Kubernetes as a non-root user with a read-only root filesystem

### DISCUSSION THREAD

**Discussion by mwyrembl** [eclipse-theia/theia#8254] [General]
Is it possible to run Theia IDE in Kubernetes as a non-root user with a read-only root filesystem?
If so, how did you manage to do it?

**Comment by any-lia** [comment]
I have been able to run Theia in Kubernetes without any root privilege, but i'm not sure you can do it with read-only filesystem because Theia need to write files, within /tmp for instance.
The default user is `theia` when you use the Docker image, filesystem access is already limited.

What's your goal ?

**Reply by mwyrembl** [reply]
Well the goal is to run theia in a Kubernetes environment with a PodSecurityPolicy that does not allow running containers as root and also enforces the root file system to be read-only. It is only possible to mount ephemeral or persistent volumes in order to write to the file system. But this will only work if files that are already part of the container image are not being modified.
